climada/hazard/tc_tracks.py

```diff
@@ -343,48 +389,59 @@ def read_ibtracs_netcdf(self, provider=None, storm_id=None,
             LOGGER.warning('No valid timestamps found for %s.', st_ids)
             ibtracs_ds = ibtracs_ds.sel(storm=valid_st)
 
-        if not provider:
-            agency_pref, track_agency_ix = ibtracs_track_agency(ibtracs_ds)
+        if provider_wind_corr:
+            for agency in IBTRACS_AGENCIES:
+                ibtracs_ds[f'{agency}_wind'] *= IBTRACS_AGENCY_1MIN_WIND_FACTOR[agency]
 
-        for var in ['wind', 'pres', 'rmw', 'poci', 'roci']:
-            if provider:
-                # enforce use of specified provider's data points
-                ibtracs_ds[var] = ibtracs_ds[f'{provider}_{var}']
-            else:
-                # array of values in order of preference
-                cols = [f'{a}_{var}' for a in agency_pref]
-                cols = [col for col in cols if col in ibtracs_ds.data_vars.keys()]
-                all_vals = ibtracs_ds[cols].to_array(dim='agency')
-                preferred_ix = all_vals.notnull().argmax(dim='agency')
-
-                if var in ['wind', 'pres']:
-                    # choice: wmo -> wmo_agency/usa_agency -> preferred
-                    ibtracs_ds[var] = ibtracs_ds['wmo_' + var] \
-                        .fillna(all_vals.isel(agency=track_agency_ix)) \
-                        .fillna(all_vals.isel(agency=preferred_ix))
-                else:
-                    ibtracs_ds[var] = all_vals.isel(agency=preferred_ix)
-        if provider:
-            # enforce use of specified provider's coordinates
-            ibtracs_ds['lat'] = ibtracs_ds[f'{provider}_lat']
-            ibtracs_ds['lon'] = ibtracs_ds[f'{provider}_lon']
+        if provider is None:
+            provider = ["official_3h"] + IBTRACS_AGENCIES
+        provider = [provider] if isinstance(provider, str) else provider
+
+        for var in ['lat', 'lon', 'wind', 'pres', 'rmw', 'poci', 'roci']:
+            if "official" in provider or "official_3h" in provider:
+                ibtracs_add_official_variable(ibtracs_ds, var, add_3h=("official_3h" in provider))
+
+            # set up dimension of agency-reported values in order of preference, including the
+            # newly created `official` and `official_3h` data if specified
+            cols = [f'{a}_{var}' for a in provider]
+            cols = [col for col in cols if col in ibtracs_ds.data_vars.keys()]
+            all_vals = ibtracs_ds[cols].to_array(dim='agency')
+            preferred_ix = all_vals.notnull().any(dim="date_time").argmax(dim='agency')
+            ibtracs_ds[var] = all_vals.isel(agency=preferred_ix)
```

Comment:
No, this is not correct. For a given variable AND a given storm, a single agency will be used. For different storms, different agencies might be used for a single variable such as `wind`. This is also explained in the updated docstring.

Reply:
Could you please also explain to me how is the interplay with the interpolation? Let me explain my question a bit better. Could the following happen?
Suppose that for 'track 1', agency 'official' is chosen for the variable 'pressure', and agency 'official_3h' for variable 'windspeed'. Then, then 'pressure' would be in 6-hours steps, while 'windspeed' in 3-hour steps. Would this not introduce inconsistencies?

Reply:
The time dimension in IBTrACS has a fixed length of 360 for all storms and agencies. So, all variables for all storms and agencies have the same length. If an entry for a particular variable, agency and time step has not been reported by that agency, it's set to NaN in IBTrACS.

For the particular example you describe: If you have `provider=['official', 'official_3h']`, it will always choose the 'official' data for every variable. So that parameter is completely equivalent to `provider='official'`. This is because 'official_3h' data will be available if and only if 'official' is available.
However, suppose we have 'pressure' in 6-hour time steps and 'windspeed' in 3-hour time steps (for some other reason). In that case, the pressure variable will have NaNs at the "unofficial" time steps, while the windspeed variable has them. If you deactivate `interpolate_missing` and `estimate_missing`, the unofficial time steps will be discarded, because only time steps where both 'windspeed' and 'pressure' are available are considered.